### PR TITLE
Update soft-crud.repository.base.ts

### LINKED ISSUE #35: Filter mixup in find function when using an or/and condition.
**Describe the bug**
In soft-crud.repository.base.ts, the find function update the 'filter' parameter to add a 'deleted:false' condition. However, in some case, this is not done correctly  and the filter is incorrectly updated...and so the final query is incorrect.

**To Reproduce**
Example of my filter:
```
in   find(filter?: Filter<T>, options?: Options): Promise<(T & Relations)[]> {
filter is filled like below
{
  order: {[ 'xxx ASC']},
  where: {
    id: {eq: 2},
    myColumn: 42,
    or: {[
       {a:1},
       {b:2}
    ]}
  }
```

The "find" function expect a where filter to start with **where:{AND**: or **where: {OR**. If the where filter doesn't start with an AND array or OR array, the filter expect an object (which is an implicit AND), and add "deleted:false" 

When the object contain an AND / OR **property**, with other properties in the object, the filter is incorrectly updated...and so the final query is incorrect.

```
in soft-crud.repository.base.ts
find(filter?: Filter<T>, options?: Options): Promise<(T & Relations)[]> {
    // Filter out soft deleted entries
    if (
      filter?.where &&
      (filter.where as AndClause<T>).and &&
      (filter.where as AndClause<T>).and.length > 0
    ) {
      (filter.where as AndClause<T>).and.push({
        deleted: false,
      } as Condition<T>);
    } else if (
      filter?.where &&
      (filter.where as OrClause<T>).or &&
      (filter.where as OrClause<T>).or.length > 0
    ) {
      filter.where = {
        and: [
          {
            deleted: false,
          } as Condition<T>,
          {
            or: (filter.where as OrClause<T>).or,
          },
        ],
      };
    } else {
      filter = filter ?? {};
      filter.where = filter.where ?? {};
      (filter.where as Condition<T>).deleted = false;
    **}**
```

**Expected behavior**
The find function should better detect when AND/OR/{} is used, and when **or** or **and** is not alone in the object.

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by arnaud16571542**
Replace buggy where manipulation, trying to impose delete:false, with FilterBuilder & WhereBuilder

## Description
Fixes #35 

## Type of change
- [X ] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested ?
Manually tested

## Checklist:
- [X ] Performed a self-review of my own code

### CODE REVIEW

File: src/repositories/soft-crud.repository.base.ts [hunk 1]
```diff
@@ -34,157 +35,42 @@ export abstract class SoftCrudRepository<
...
-    return super.findOne(filter, options);
-  }
-
-  //findOne() including soft deleted entry
-  findOneIncludeSoftDelete(
-    filter?: Filter<T>,
-    options?: Options,
-  ): Promise<(T & Relations) | null> {
-    return super.findOne(filter, options);
+    return super.findOne(fb.build(), options);
```
Comment (akshatdubeysf) [inline src/repositories/soft-crud.repository.base.ts:54]: @arnaud16571542 The findOneIncludeSoftDelete is missing?

File: src/repositories/soft-crud.repository.base.ts [hunk 2]
```diff
@@ -34,157 +35,42 @@ export abstract class SoftCrudRepository<
...
-    }
 
-    // Now call super
-    return super.find(filter, options);
-  }
+    let fb = new FilterBuilder<T>(filter)
+    fb.impose({deleted:false} as Where<T>)
 
-  //find all enteries including soft deleted records
-  findAll(filter?: Filter<T>, options?: Options): Promise<(T & Relations)[]> {
```
Comment (samarpan-b) [inline src/repositories/soft-crud.repository.base.ts:72]: need this method

File: src/repositories/soft-crud.repository.base.ts [hunk 3]
```diff
@@ -34,157 +35,42 @@ export abstract class SoftCrudRepository<
...
       // Now call super
-      return super.findById(id, filter, options);
+      return super.findById(id, fb.build(), options);
     } else {
-      throw new HttpErrors.NotFound(ErrorKeys.EntityNotFound);
-    }
-  }
-
-  //find by Id including soft deleted record
-  async findByIdIncludeSoftDelete(
```
Comment (samarpan-b) [inline src/repositories/soft-crud.repository.base.ts:174]: need this method

**Review by samarpan-b — CHANGES_REQUESTED** [review]

**Comment by akshatdubeysf** [comment]
This has been implemented through another PR.

**Pull request closed without merging**